test/models/test_cice.py

```diff
@@ -300,3 +307,85 @@ def test_no_restart_ptr(config, cice_config_files):
         with pytest.raises(RuntimeError,
                            match="Cannot find previous restart file"):
             model.setup()
+
+
+CONFIG_WITH_COMPRESSION = {
+    "laboratory": "lab",
+    "jobname": "testrun",
+    "model": "cice",
+    "exe": "test.exe",
+    "experiment": ctrldir_basename,
+    "metadata": {"enable": False},
+    "compress_logs": True
+}
+
+
+@pytest.fixture
+def cice4_log_files():
+    """
+    Create cice log files matching those produced during ESM1.5 simulations.
+    """
+    log_names = ["ice_diag_out", "ice_diag.d", "debug.root.03",
+                 "iceout086", "iceout088", "iceout090", "iceout092",
+                 "iceout094", "iceout096", "iceout085", "iceout087",
+                 "iceout089", "iceout091", "iceout093", "iceout095"]
+    log_paths = [os.path.join(expt_workdir, name) for name in log_names]
+
+    for log_file in log_paths:
+        with open(log_file, "w") as f:
+            f.close()
+
+    yield log_paths
+
+    # Cleanup
+    for log_file in log_paths:
+        try:
+            os.remove(log_file)
+        except FileNotFoundError:
+            pass
+
+
+@pytest.fixture
+def non_log_file():
+    """
+    Create a cice4 output file to be ignored by log compression.
+    Use cice_in.nml which is copied to the work directory in ESM1.5.
+    """
+    non_log_path = os.path.join(expt_workdir, CICE_NML_NAME)
+    with open(non_log_path, "w") as f:
+        f.close()
+
+    yield non_log_path
+
+    # Cleanup
+    os.remove(non_log_path)
+
+
+@pytest.mark.parametrize("config", [CONFIG_WITH_COMPRESSION],
+                         indirect=True)
+def test_log_compression(config, cice4_log_files, non_log_file,
+                         cice_nml   # Required by expt.__init__
+                         ):
+    """
+    Test that logfiles produced by cice during ESM1.5 simulations are
+    properly compressed into a tarball by cice.compress_log_files().
+    """
+    with cd(ctrldir):
+        # Initialise laboratory and experiment
+        lab = payu.laboratory.Laboratory(lab_path=str(labdir))
+        expt = payu.experiment.Experiment(lab, reproduce=False)
+        model = expt.models[0]
+
+        # Function to test
+        model.compress_log_files()
```

Comment:
I originally hoped to test `model.archive()` here, which would allow us to test the cases where `compress_logs` is and isn't included in the `config.yaml`. However running the actual `cice.archive()` required several directories and files to be created that weren't really relevant to what I wanted to test, and I think they would have made the added tests fairly brittle. Hence I ended up just testing the `compress_log_files()` method directly.

If we move everything over to the `model.py` driver, I think it will be easier to run a more general test of `model.archive()` as there would be less requirements on the work directory.

Reply:
Definitely worth doing in the future, but I'd hate to hold this up any further.